server/index.js

```diff
@@ -90,9 +90,70 @@ app.post(('/api/collection'), (req, res, next) => {
   const dbQuery = db.query(sql, params);
   dbQuery.then(result => {
     res.status(201).send(result.rows[0]);
-  }).catch(err => {
-    next(err);
-  });
+  }).catch(err => next(err));
+});
+
+app.post(('/api/restaurant'), (req, res, next) => {
+  const { collectionId, businessId } = req.body;
+  if (!collectionId || !businessId) {
+    throw new ClientError(400, 'collectionId and businessId are required');
+  }
+  client.business(businessId)
+    .then(response => {
+      const json = response.jsonBody;
+      const sql = `
+      insert into "restaurants" ("collectionId", "businessId", "json")
+      values ($1, $2, $3)
+      returning *;
+      `;
+      const params = [collectionId, businessId, json];
+      const dbQuery = db.query(sql, params);
+      dbQuery.then(result => {
+        res.status(201).send(result.rows[0]);
+      }).catch(err => next(err));
+      const setImageSQL = `
+        update "collections"
+           set "image" = coalesce("image", $1)
+         where "collectionId" = $2
+      `;
+      const setImageParams = [json.image_url, collectionId];
+      const dbQueryImage = db.query(setImageSQL, setImageParams);
+      dbQueryImage.then()
+        .catch(err => next(err));
+    })
+    .catch(err => next(err));
+});
+
+app.get(('/api/collections'), (req, res, next) => {
+  const sql = `
+  select "c"."name",
+         "c"."collectionId",
+         "c"."image",
+         count ("r"."collectionId")
+    from "collections" as "c"
+    left join "restaurants" as "r" on "r"."collectionId" = "c"."collectionId"
+   where "userId" = $1
+   group by "c"."collectionId";
+  `;
+  const params = [1];
+  const dbQuery = db.query(sql, params);
+  dbQuery.then(result => {
+    res.status(200).send(result.rows);
+  }).catch(err => next(err));
+});
+
+app.get(('/api/restaurant'), (req, res, next) => {
+  const sql = `
+  select "businessId"
+    from "restaurants";
+  `;
+  const dbQuery = db.query(sql);
+  dbQuery.then(result => {
+    const businessIds = result.rows.map(businessId => {
+      return businessId.businessId;
+    });
```

Comment:
your `.map()` callback parameter name is misleading here. whenever you use `.map()`, the parameter to the callback is the singular form of the array's name.

instead of this

```js
result.rows.map(businessId => {
```

do this

```js
result.rows.map(row => {
```